packages/manager/core/generator/app/templates/src/components/Breadcrumb.tsx

```diff
@@ -21,43 +28,35 @@ type Match = {
   params: Params<string>;
 };
 
-export default function BreadcrumbComponent(): JSX.Element {
+function Breadcrumb(): JSX.Element {
   const matches = useMatches();
-  const [crumbs, setCrumbs] = useState<Record<string, string>[]>();
+  const [crumbs, setCrumbs] = useState<BreadcrumbItem[]>([]);
 
   useEffect(() => {
     const items = matches.map(async (match) => {
       const { handle, data, params }: Match = match;
       const breadcrumb = await handle?.breadcrumb;
-      return {
-        pathname: match.pathname,
-        crumb: await breadcrumb?.({ data, params }),
+      const crumb = {
+        label: await breadcrumb?.({ data, params }),
+        href: match.pathname,
       };
+      return crumb;
     });
 
-    Promise.all(items)
-      .then((breadcrumbs) => {
-        setCrumbs(breadcrumbs.filter(({ crumb }) => !!crumb));
-      })
-      .catch((err) => console.error(err));
+    Promise.all(items).then((breadcrumbs) => {
+      const validCrumbs = breadcrumbs.filter(
+        ({ label, href }) => !!label && !!href,
+      );
+      setCrumbs(validCrumbs);
+    });
   }, [matches]);
 
   // no need to display a breadcrumb for a single item
-  if (!crumbs || crumbs.length <= 1) {
+  if (crumbs.length <= 1) {
     return <></>;
   }
 
-  return (
-    <ul>
-      {crumbs.map(({ pathname, crumb }, index) => (
-        <li key={pathname}>
-          {index + 1 < crumbs.length ? (
-            <Link to={pathname}>{crumb}</Link>
-          ) : (
-            <span>{crumb}</span>
-          )}
-        </li>
-      ))}
-    </ul>
-  );
+  return <OsdsBreadcrumb items={crumbs} />;
```

Comment:
```suggestion
  return crumbs?.length <= 1 ? <></> : <OsdsBreadcrumb items={crumbs} />;
```

We can merge the 2 lines with a ternary